draft-ietf-httpbis-digest-headers.md

```diff
@@ -281,70 +319,112 @@ that the recipient will ignore it.
+
+~~~ http-message
+Representation-Digest: sha-512=:WZDPaVn/7XgHaAy8pmojAkGWoRx2UFChF41A2svX+TaPm
+                       AbwAgBWnrIiYllu7BNNyealdVLvRwE\nmTHWXvJwew==:
+~~~
+
+Since the `Representation-Digest` field is a Dictionary, it can contain multiple
+members. This could be used, for example, to attach multiple checksums
+calculated using different hashing algorithms in order to support a population
+of endpoints with different evolving capabilities. Such an approach could
+support transitions away from weaker algorithms (see {{algorithm-agility}}).
+
+~~~ http-message
+Representation-Digest: sha-256=:4REjxQ4yrqUVicfSKYNO/cF9zNj5ANbzgDZt3/h3Qxo=:,
+                       sha-512=:WZDPaVn/7XgHaAy8pmojAkGWoRx2UFChF41A2svX+TaPm
+                       AbwAgBWnrIiYllu7BNNyealdVLvRwE\nmTHWXvJwew==:
+~~~
+
+A recipient MAY ignore any or all of members of the Representation-Digest field.
+This allows the recipient to choose which hashing algorithm(s) to use for
+validation instead of verifying every received representation-data-digest.
+
+A sender MAY send a `Representation-Digest` member without knowing whether the
+recipient supports a given hashing algorithm, or even knowing that the recipient
+will ignore it.
 
+`Representation-Digest` can be sent in a trailer section.
+In this case,
+`Representation-Digest` MAY be merged into the header section; see {{Section 6.5.1 of SEMANTICS}}.
 
-# The Content-Digest Field {#content-digest}
+# The Content-Digest Field {#content-digest-field}
 
-The `Content-Digest` field contains a comma-separated list of one or more content digest
-values.
-A content digest value is computed by applying a digest-algorithm to the actual message content
-(see {{Section 6.4 of SEMANTICS}}).
-It can be used in both requests and responses.
+The `Content-Digest` HTTP field can be used in requests and responses to
+communicate checksums that are calculated using a hashing algorithm applied to
+the actual message content (see {{Section 6.4 of SEMANTICS}}). It is a
+Structured Fields Dictionary (see {{Section 3.2 of STRUCTURED-FIELDS}}):
 
 ~~~ abnf
-   Content-Digest = 1#content-digest
-   content-digest = digest-algorithm "="
-                    <encoded checksum output>
+Content-Digest   = sf-dictionary
 ~~~
 
+`Content-Digest` member-keys convey the hashing algorithm (see {{algorithms}})
+used to compute the checksum. Member-values are the the output of the checksum
+calculation. Member-values MUST be of type sf-binary.
+
 For example:
 
 ~~~ http-message
-Content-Digest: sha-512=WZDPaVn/7XgHaAy8pmojAkGWoRx2UFChF41A2svX+TaPm
-                        AbwAgBWnrIiYllu7BNNyealdVLvRwE\nmTHWXvJwew==
+Content-Digest: sha-512=:WZDPaVn/7XgHaAy8pmojAkGWoRx2UFChF41A2svX+TaPm
+                        AbwAgBWnrIiYllu7BNNyealdVLvRwE\nmTHWXvJwew==:
 ~~~
 
-A `Content-Digest` field MAY contain multiple content-digest values,
-similarly to `Digest` (see {{digest}})
+Since the `Content-Digest` field is a Dictionary, it can contain multiple
```

Comment:
@LPardue since the original document, this paragraph is repeated multiple times. Considered that here https://github.com/httpwg/http-extensions/pull/1893/files#r793058641 the `MAY` is redundant, do you think we can just state this sentence once and for all somewhere?

Reply:
I think we can deduplicate text. With headers it was bad enough with 3 it gets worse.

But I'm holding off investing time on deep editorial work until there is consensus to adopt or reject either proposal

Reply:
Just tracking down stuff, here